colorizer_data/types.py

```diff
@@ -97,15 +100,64 @@ class FrameDimensions(TypedDict):
     """Height of a frame in physical units (not pixels)."""
 
 
-class DatasetMetadata(TypedDict):
+class BaseMetadataJson(TypedDict):
+    """JSON dictionary format for `BaseMetadata`."""
+
+    name: str
+    description: str
+    author: str
+    dateCreated: str
+    lastModified: str
+    revision: str
+    dataVersion: str
+
+
+@dataclass
+class BaseMetadata:
+    """Shared metadata between datasets and collection files."""
```

Comment:
I suggest not sharing this at all and letting it be just duplicated because it's much easier to read and it easily allows for them to diverge.

Reply:
As I am thinking about this, I wonder if we could always force there to be a "collection".